submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/service/StorageService.kt

```diff
@@ -49,7 +49,20 @@ class StorageService(
         }
     }
 
-    override fun deleteSubmissionFiles(sub: ExtSubmission) {
-        serializationService.fileSequence(sub).forEach { file -> deleteSubmissionFile(sub, file) }
+    override fun deleteSubmissionFiles(
+        sub: ExtSubmission,
+        onEachIndexed: (Int, ExtFile) -> Unit,
+        filter: (ExtFile) -> Boolean,
+    ) {
+        serializationService.fileSequence(sub)
```

Comment:
Here we add the logging since it's what is actually doing, deleting

(Continues below 2/3)